message/ipldcbor/schema.ipldsch

```diff
@@ -0,0 +1,78 @@
+# TODO(mvdan): upstream must support Any sooner than later
+type Any union {
+	| Bool   bool
+	| Int    int
+	| Float  float
+	| String string
+	| Bytes  bytes
+	# TODO(mvdan): these are missing from upstream because of Any, too
+	# | Map    map
+	# | List   list
+	# | Link   link
+} representation kinded
+
+type GraphSyncExtensions {String:Any}
+type GraphSyncRequestID int
+type GraphSyncPriority int
+
+type GraphSyncMetadatum struct {
+  link         Link # TODO(mvdan): support "&" links in the ipld-prime schema loader
+  blockPresent Bool
+} representation tuple
+
+type GraphSyncMetadata [GraphSyncMetadatum]
+
+type GraphSyncResponseCode enum {
+  # Informational Codes (request in progress)
+
+  | RequestAcknowledged ("10")
+  | AdditionalPeers ("11")
+  | NotEnoughGas ("12")
+  | OtherProtocol ("13")
+  | PartialResponse ("14")
+  | RequestPaused ("15")
+
+  # Success Response Codes (request terminated)
+
+  | RequestCompletedFull ("20")
+  | RequestCompletedPartial ("21")
+
+  # Error Response Codes (request terminated)
+
+  | RequestRejected ("30")
+  | RequestFailedBusy ("31")
+  | RequestFailedUnknown ("32")
+  | RequestFailedLegal ("33")
+  | RequestFailedContentNotFound ("34")
+  | RequestCancelled ("35")
+} representation int
+
+type GraphSyncRequest struct {
+  id          GraphSyncRequestID  (rename "ID")   # unique id set on the requester side
+  root        Link                (rename "Root") # a CID for the root node in the query
+  # TODO(mvdan): must we copy-paste the selector schema here?
+  # selector    Selector            (rename "Sel")  # see https://github.com/ipld/specs/blob/master/selectors/selectors.md
+  extensions  GraphSyncExtensions (rename "Ext")  # side channel information
+  priority    GraphSyncPriority   (rename "Pri")  # the priority (normalized). default to 1
+  cancel      Bool                (rename "Canc") # whether this cancels a request
+  update      Bool                (rename "Updt") # whether this is an update to an in progress request
+} representation map
+
+type GraphSyncResponse struct {
+  id          GraphSyncRequestID          (rename "ID")   # the request id we are responding to
+  # TODO(mvdan): this type appears to be missing
```

Comment:
FYI @hannahhoward, was this just a typo with GraphSyncResponseCode? Which of the two names do you prefer?

Reply:
Ah, found the typo, from Code to StatusCode.